### PR TITLE
tests(cypress): Refactor reconnect test to be more reliable and add second test for actual reconnect

### PR DESCRIPTION
**Pull request by juliushaertl**
This should make our top failure cypress test on main more reliable and also extends it with another case:

- The existing test only tests if an existing session is picket up after a short interrupt of sync/push requests
- The new test actually waits until the y.js websocket connection gets closed and checks if a full reconnect is working

### CODE REVIEW

**Review by mejo- — APPROVED** [review]
Awesome, thanks for tackling this :sparkles:

File: cypress/e2e/sync.spec.js [hunk 1]
```diff
@@ -64,23 +64,23 @@ describe('Sync', () => {
...
-			.as('createSession')
-		cy.intercept({ method: 'POST', url: '**/apps/text/session/*' }, (req) => {
-			if (count < 4) {
+	it('recovers from a short lost connection', () => {
+		let reconnect = false
+		cy.intercept('**/apps/text/session/*', (req) => {
+			if (reconnect) {
+				req.continue()
+				req.alias = 'alive'
+				reconnect = true
```
Comment (max-nextcloud) [inline cypress/e2e/sync.spec.js:73]: This seems somewhat redundant as reconnect is already checked to be truethy in line 70

Or am I missing something?

**Review by max-nextcloud — APPROVED** [review]
Yay! Thanks for getting to the bottom of this! :diving_mask: 
Also great idea to test the close / create loop.

I have just one small remark.

**Comment by max-nextcloud** [comment]
Maybe also backport to stable27...
grafik

**Pull request merged**